packages/datadog-plugin-http/src/client.js

```diff
@@ -145,45 +145,80 @@ function patch (http, methodName, tracer, config) {
     })
   }
 
-  function extractUrl (options) {
-    const uri = options
-    const agent = options.agent || http.globalAgent
-
-    return typeof uri === 'string' ? uri : url.format({
-      protocol: options.protocol || agent.protocol,
-      hostname: options.hostname || options.host || 'localhost',
-      port: options.port,
-      pathname: options.path || options.pathname || '/'
-    })
-  }
-
   function normalizeArgs (inputURL, inputOptions, callback) {
-    let options = normalizeURL(inputURL)
+    inputURL = normalizeOptions(inputURL)
+
+    const [callbackNormalilzed, inputOptionsNormalized] = normalizeCallback(inputOptions, callback, inputURL)
+    const coalesedOptions = coaleseOptions(inputURL, inputOptionsNormalized)
+    const uri = url.format(coalesedOptions)
 
+    return { uri: uri, options: coalesedOptions, callback: callbackNormalilzed }
```

Comment:
FWIW I would probably have used `uri`, `options` and `callback` as local variable names since that's what is returned which would make the return value easier to write/read.

Reply:
makes sense to me, updated